tests.js

```diff
@@ -9,9 +9,47 @@
     return;
   }
 
-  function qpair(template, ...values) {
-    return [template, values];
+  // Set up a RegExp subclass so that we can test subclass creation.
+  function SubRegExp(source, flags) {
+    RegExp.call(this, source, flags);
```

Comment:
I did the first.  Instead of doing  `Object.setPrototypeOf(SubRegExp, RegExp);` I modified the example ljharb pointed to, specifically https://github.com/paulmillr/es6-shim/blob/master/test/collections.js#L137